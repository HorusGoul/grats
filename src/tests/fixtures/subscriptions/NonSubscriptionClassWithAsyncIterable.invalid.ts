/** @gqlType */
export class User {
  /** @gqlField */
  async *greetings(): AsyncIterable<string> {
    yield "Hello";
    yield "World";
  }
}
