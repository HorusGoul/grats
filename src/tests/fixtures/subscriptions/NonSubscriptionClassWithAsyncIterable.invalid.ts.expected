-----------------
INPUT
----------------- 
/** @gqlType */
export class User {
  /** @gqlField */
  async *greetings(): AsyncIterable<string> {
    yield "Hello";
    yield "World";
  }
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/subscriptions/NonSubscriptionClassWithAsyncIterable.invalid.ts:4:23 - error: Unexpected AsyncIterable. Only fields on `Subscription` should return an `AsyncIterable`. Non-subscription fields are only expected to return a single value.

4   async *greetings(): AsyncIterable<string> {
                        ~~~~~~~~~~~~~~~~~~~~~
